donate/payments/forms.py

```diff
@@ -48,6 +48,31 @@ def clean(self):
                 })
 
 
+class PostalCodeMixin():
+    """
+    Mixin for checking whether or not we should require the postcode
+    for a payment from the donate page.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def clean(self):
+        cleaned_data = super().clean()
+        # Grabbing post code and selected county from form data
+        postal_code = cleaned_data.get('post_code', '')
+        country = cleaned_data.get('country', '')
+        # Getting countries post code information from list in constants.py
+        country_object_to_check = next(country_obj for country_obj in
+                                       constants.COUNTRY_POST_CODES
+                                       if country_obj["abbrev"] == country)
+        # If country uses a post code and the user left it blank, raise a validation error.
+        if 'postal' in country_object_to_check and postal_code == "":
+            raise forms.ValidationError({
+                'post_code': _('This field is required.')
+            })
```

Comment:
Remember the general rule for comments: if the comment says the same thing that the code does, it doesn't need to be a comment. In this case you may want a comment to explain why the coded uses `next(...)` but the rest of the code is clear enough on its own.